doc/SAI-Packet-Action-Group.md

```diff
@@ -0,0 +1,729 @@
+        _Out_ sai_object_id_t *packet_action_group_member_id,
+        _In_ sai_object_id_t switch_id,
+        _In_ uint32_t attr_count,
+        _In_ const sai_attribute_t *attr_list);
+
+/**
+ * @brief Remove packet action group member
+ *
+ * @param[in] packet_action_group_member_id Packet action group member ID
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_remove_packet_action_group_member_fn)(
+        _In_ sai_object_id_t packet_action_group_member_id);
+
+/**
+ * @brief Set packet action group member attribute
+ *
+ * @param[in] packet_action_group_member_id Packet action group member ID
+ * @param[in] attr Attribute
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_set_packet_action_group_member_attribute_fn)(
+        _In_ sai_object_id_t packet_action_group_member_id,
+        _In_ const sai_attribute_t *attr);
+
+/**
+ * @brief Get packet action group member attribute
+ *
+ * @param[in] packet_action_group_member_id Packet action group member ID
+ * @param[in] attr_count Number of attributes
+ * @param[inout] attr_list Array of attributes
+ *
+ * @return #SAI_STATUS_SUCCESS on success, failure status code on error
+ */
+typedef sai_status_t (*sai_get_packet_action_group_member_attribute_fn)(
+        _In_ sai_object_id_t packet_action_group_member_id,
+        _In_ uint32_t attr_count,
+        _Inout_ sai_attribute_t *attr_list);
+
+/**
+ * @brief Packet action group methods table retrieved with sai_api_query()
+ */
+typedef struct _sai_packet_action_group_api_t
+{
+    sai_create_packet_action_group_fn               create_packet_action_group;
+    sai_remove_packet_action_group_fn               remove_packet_action_group;
+    sai_set_packet_action_group_attribute_fn        set_packet_action_group_attribute;
+    sai_get_packet_action_group_attribute_fn        get_packet_action_group_attribute;
+    sai_create_packet_action_group_member_fn        create_packet_action_group_member;
+    sai_remove_packet_action_group_member_fn        remove_packet_action_group_member;
+    sai_set_packet_action_group_member_attribute_fn set_packet_action_group_member_attribute;
+    sai_get_packet_action_group_member_attribute_fn get_packet_action_group_member_attribute;
+
+} sai_packet_action_group_api_t;
+
+```
+
+##### Changes to existing SAI header file
+**In sai.h**
+1. New object for SAI Packet Action Group
+
+```
+    SAI_API_ISOLATION_GROUP     = 40, /**< sai_isolation_group_api_t */
++   SAI_API_PACKET_ACTION_GROUP = 41, /**< sai_packet_action_group_api_t */
+    SAI_API_MAX                 = 42, /**< total number of APIs */
+} sai_api_t;
+
+```
+
+**In saitypes.h**
+1. Add new structure for packet action group actions
+2. Include the new structure in the sai_attribute_value_t structure
+
+```
+/**
+ * @brief Packet Action Group Action Type
+ */
+typedef enum _sai_packet_action_group_action_type_t
```

Comment:
Another idea could be to define a packet modification object. The attributes of the object could be the list fields to be enabled and their values. The OID can be used as action MODIFY. This gives a way to use the same set of modification for multiple groups and we can use it as direct ACL action too.